wallet/wallet/hd_wallet.go

```diff
@@ -326,6 +326,11 @@ func (w *HDWallet) MarshalJSON() ([]byte, error) {
 		Keys:        w.keyRing.ListKeyPairs(),
 		Permissions: w.permissions,
 	}
+
+	if jsonW.Permissions == nil {
```

Comment:
how can jsonW.Permissions be nil? it's a map not a pointer to a map isn't it?

Reply:
AFAIK, well that was my mistake. I never though map could be nil.

https://go.dev/play/p/57tl546Uth3

Seems to be possible.

Reply:
your playground is not what's happening in your code though and doesn't explain why you needed this change. This is more like it: https://go.dev/play/p/y6BHV9NYOvu

Reply:
but to be fair `tStruct.testMap == nil` evaluates to true. so you're right.